pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -135,29 +199,131 @@ def __init__(
             c_dtype=np.complex128,
             shots=None,
             batch_obs: Union[bool, int] = False,
+            analytic=None,
         ):
-            super().__init__(wires, c_dtype=c_dtype, shots=shots)
-            self._gpu_state = _gpu_dtype(self._state.dtype)(self._state)
+            if c_dtype is np.complex64:
+                r_dtype = np.float32
+                self.use_csingle = True
+            elif c_dtype is np.complex128:
+                r_dtype = np.float64
+                self.use_csingle = False
+            else:
+                raise TypeError(f"Unsupported complex Type: {c_dtype}")
+            super().__init__(
+                wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype, analytic=analytic
+            )
+            self._gpu_state = _gpu_dtype(c_dtype)(self.num_wires)
+            self._create_basis_state_GPU(0)
             self._sync = sync
             self._dp = DevPool()
             self._batch_obs = batch_obs
 
         def reset(self):
             super().reset()
+            # init the state vector to |00..0>
             self._gpu_state.resetGPU(False)  # Sync reset
 
-        def syncH2D(self, use_async=False):
+        def syncD2H(self, state_vector, use_async=False):
+            """Explicitly synchronize GPU data to CPU"""
+            self._gpu_state.DeviceToHost(state_vector.ravel(order="C"), use_async)
+
+        def syncH2D(self, state_vector, use_async=False):
             """Explicitly synchronize CPU data to GPU"""
-            self._gpu_state.HostToDevice(self._state.ravel(order="C"), use_async)
+            self._gpu_state.HostToDevice(state_vector.ravel(order="C"), use_async)
 
-        def syncD2H(self, use_async=False):
-            """Explicitly synchronize GPU data to CPU"""
-            self._gpu_state.DeviceToHost(self._state.ravel(order="C"), use_async)
-            self._pre_rotated_state = self._state
+        def _create_basis_state_GPU(self, index, use_async=False):
+            """Direct set the 'index'th element on GPU
+            Args:
+                index (int): The index of element of statevector to be set.
+            """
+            self._gpu_state.setBasisState(index, use_async)
+
+        def _apply_state_vector_GPU(self, state, device_wires, use_async=False):
+            """Initialize the internal state vector in a specified state on GPU.
+            Args:
+            state (array[complex]): normalized input state of length ``2**len(wires)``
+                 or broadcasted state of shape ``(batch_size, 2**len(wires))``
+            device_wires (Wires): wires that get initialized in the state
+            """
+            # translate to wire labels used by device
+            device_wires = self.map_wires(device_wires)
+            dim = 2 ** len(device_wires)
+
+            state = self._asarray(state, dtype=self.C_DTYPE)
+            batch_size = self._get_batch_size(state, (dim,), dim)
+            output_shape = [2] * self.num_wires
```

Comment:
Since this assumes the creation of a host array, we should also mention that in the docstring. Any use of this method will introduce host-overheads.

Best to be explicit, or people will wonder why there is a sync needed.